jms/deployment/src/main/java/io/quarkus/artemis/jms/deployment/ArtemisJmsProcessor.java

```diff
@@ -108,7 +109,8 @@ private static SyntheticBeanBuildItem toSyntheticBeanBuildItem(
     private static SyntheticBeanBuildItem.ExtendedBeanConfigurator initializeConfigurator(boolean isXaEnable) {
         SyntheticBeanBuildItem.ExtendedBeanConfigurator configurator;
         if (isXaEnable) {
-            configurator = SyntheticBeanBuildItem.configure(ActiveMQConnectionFactory.class);
+            configurator = SyntheticBeanBuildItem.configure(ActiveMQXAConnectionFactory.class);
+            configurator.addType(ActiveMQConnectionFactory.class);
```

Comment:
The [ActiveMQConnectionFactory](https://github.com/apache/activemq-artemis/blob/main/artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java#L69) implements both ConnectionFactory and XAConnectionFactory interface. So I have no strong opinion to use ActiveMQXAConnectionFactory here.